core/src/main/java/com/spotify/metrics/core/Distribution.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (c) 2016 Spotify AB.
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.spotify.metrics.core;
+
+import com.codahale.metrics.Counting;
+import com.codahale.metrics.Metric;
+
+import java.nio.ByteBuffer;
+
+/**
+ * {@link Distribution} is a simple interface that allows users to record measurements
+ * to compute rank statistics on data distribution not just local source.
+ *
+ * <p>Unlike traditional histogram, {@link Distribution} doesn't require
+ * predefined percentile value. Data recorded
+ * can be used upstream to compute any percentile.
+ *
+ * <p>This Distribution doesn't require any binning configuration.
+ * Just get an instance through SemanticMetricBuilder and record data.
+ *
+ * <p> {@link Distribution} is a good choice if you care about percentile accuracy in
+ * a distributed environment and you want to rely on P99 to set SLO.
+ */
+public interface Distribution extends Metric, Counting {
+
+    /**
+     * Record value from Min.Double to Max.Double.
+     * @param val
+     */
+    void record(double val);
+
+    /**
+     * Return distribution point value and flush.
+     * When this method is called every internal state
+     * is reset and a new recording starts.
+     * @return
+     */
+    ByteBuffer getValueAndFlush();
```

Comment:
I am still curious about the idea of exposing the "value" of the Distribution as a `ByteBuffer`. It seems like it would could this `Distribution` metric to the way in which the t-digest library serializes the underly `com.tdunning.math.stats.TDigest` into bytes.

For instance, if there was an alternate implementation of `Distribution` in the future, it would have to return the values in the same byte format, since `FastForwardReporter` has the expectation that the results of `getValueAndFlush()` must look a certain way.

Is there any other data type that could be used? Or at least could the format of the bytes be documented here?

Reply:
If we use Distribution as the return type then the serializer will have to know something about Distribution implementation. For instance, FFW reporter will have an implementation of this method for every Distribution type. 
https://github.com/spotify/semantic-metrics/pull/79/files/4ae53fe81b12e35376529ab79184798e982e7ade#diff-455c8027b591558e13ad0a4d245b2518R265. That will not align with the current paradigm of not exposing the detail of the metric implementation to the serializer.

Documentation   bytes format is handled by FFW.  There is a version number, we will use that number upstream to determine how to handle the bytes.

Reply:
1. Return something immutable like ByteString
2. Instead of ByteString, return something more semantically meaningful like a DistributionSnapshot wrapper

Reply:
1. I am not sure about DistributionSnapshot, that will expose Distribution implementation to ffwReporter as noted above.
2.  Let's  merge this branch.  Bytebuffer is transformed  into ByteString  before serialization in ffw-client.  I will  change the return type later.